lib/screenplay/config/config.ex

```diff
@@ -1,12 +1,45 @@
 defmodule Screenplay.Config.PermanentConfig do
   @moduledoc false
+  alias Screenplay.Config.S3Fetch
 
-  def add_new_screen do
+  @spec add_new_screen(binary(), map(), binary()) :: :error | :ok
+  def add_new_screen(screen_id, screen, etag) do
+    case get_current_config(etag) do
+      {:ok, config} ->
+        config
+        |> Map.put(screen_id, Jason.decode!(screen))
+        |> S3Fetch.put_screens_config()
+
+      _ ->
+        :error
+    end
   end
 
   def edit_screen do
   end
 
-  def delete_screen do
+  @spec delete_screen(binary(), binary()) :: :error | :ok
+  def delete_screen(screen_id, etag) do
+    case get_current_config(etag) do
+      {:ok, config} ->
+        config
+        |> Map.delete(screen_id)
+        |> S3Fetch.put_screens_config()
+
+      _ ->
+        :error
+    end
+  end
+
+  defp get_current_config(etag) do
+    case S3Fetch.get_screens_config() do
+      {:ok, config, current_etag} ->
+        if etag == current_etag,
+          do: {:ok, config},
+          else: :error
```

Comment:
Should be able to reduce this a little bit with more strict pattern matching in the `case`. 

```suggestion
      {:ok, config, ^etag} -> {:ok, config}
```